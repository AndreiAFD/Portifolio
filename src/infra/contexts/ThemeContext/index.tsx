import React, { createContext, useState } from "react";
import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import { ThemeProvider } from "styled-components";
import { theme } from "../../../core/theme/theme";

enum ThemeEnum {
  light = "light",
  dark = "dark",
}

interface AppThemeContextProps {
  theme?: string;
  handleChangeTheme?: (theme: ThemeEnum) => void;
}

export const AppThemeContext = createContext<AppThemeContextProps>({});

export const AppThemeContextProvider = ({ children }: { children: any }) => {
  const [actualTheme, setActualTheme] = useState<ThemeEnum>(ThemeEnum.light);

  const chakraTheme = extendTheme({
    colors: {
      ...theme[actualTheme],
    },
  });

  const handleChangeTheme = (nextTheme: ThemeEnum) => {
    setActualTheme(nextTheme);
  };

  return (
    <AppThemeContext.Provider
      value={{
        theme: actualTheme,
        handleChangeTheme,
      }}
    >
      <ThemeProvider theme={theme[actualTheme] as typeof theme["dark"]}>
        <ChakraProvider theme={chakraTheme}>{children}</ChakraProvider>
      </ThemeProvider>
    </AppThemeContext.Provider>
  );
};
