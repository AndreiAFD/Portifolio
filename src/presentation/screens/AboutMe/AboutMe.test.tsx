import { render, screen } from "@testing-library/react";
import React from "react";
import { BrowserRouter } from "react-router-dom";
import { AboutMe } from ".";
import { AppThemeContextProvider } from "../../../infra/contexts/ThemeContext";

const ComponentToTest = () => {
  return (
    <BrowserRouter>
      <AppThemeContextProvider>
        <AboutMe />
      </AppThemeContextProvider>
    </BrowserRouter>
  );
};

describe("AboutMe screen", () => {
  it("should render correctly", () => {
    render(<ComponentToTest />);
    const container = screen.getByTestId("aboutme-container-testid");
    expect(container).toMatchSnapshot();
  });
});
