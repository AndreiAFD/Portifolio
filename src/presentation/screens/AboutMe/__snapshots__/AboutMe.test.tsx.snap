// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutMe screen should render correctly 1`] = `
<div
  class="css-evdbgy"
  data-testid="aboutme-container-testid"
>
  <div
    class="css-17xy7xo"
  >
    <p
      class="chakra-text css-fbgbxk"
    >
      Quem sou eu ?
    </p>
  </div>
  <div>
    <div
      style="transition: opacity 400ms, transform 400ms; transform: translateY(20px); opacity: 0;"
    >
      <p
        class="chakra-text css-xv4nph"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam varius ac libero a rhoncus. Praesent nibh nunc, accumsan eget sollicitudin et, molestie at mauris. In placerat, tellus in congue fermentum, neque elit feugiat orci, eu pretium leo est eget lorem.
      </p>
    </div>
    <div
      style="transition: opacity 400ms, transform 400ms; transform: translateY(20px); opacity: 0;"
    >
      <p
        class="chakra-text css-s7on4h"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam varius ac libero a rhoncus. Praesent nibh nunc, accumsan eget sollicitudin et, molestie at mauris. In placerat, tellus in congue fermentum, neque elit feugiat orci, eu pretium leo est eget lorem.
      </p>
    </div>
    <div
      style="transition: opacity 400ms, transform 400ms; transform: translateY(20px); opacity: 0;"
    >
      <div
        class="css-17xy7xo"
      >
        <p
          class="chakra-text css-fbgbxk"
        >
          Principais Techs
        </p>
      </div>
    </div>
    <div
      style="transition: opacity 400ms, transform 400ms; transform: translateY(20px); opacity: 0;"
    >
      <div
        class="css-11n51jl"
      >
        <div>
          <div
            style="transition: opacity 400ms, transform 400ms; transform: translateY(20px); opacity: 0;"
          >
            <p
              class="chakra-text css-kmn13v"
              data-testid="badge-text-testid"
            >
              React
            </p>
          </div>
        </div>
        <div>
          <div
            style="transition: opacity 400ms, transform 400ms; transform: translateY(20px); opacity: 0;"
          >
            <p
              class="chakra-text css-63qbpm"
              data-testid="badge-text-testid"
            >
              React Native
            </p>
          </div>
        </div>
        <div>
          <div
            style="transition: opacity 400ms, transform 400ms; transform: translateY(20px); opacity: 0;"
          >
            <p
              class="chakra-text css-kmn13v"
              data-testid="badge-text-testid"
            >
              Firebase
            </p>
          </div>
        </div>
        <div>
          <div
            style="transition: opacity 400ms, transform 400ms; transform: translateY(20px); opacity: 0;"
          >
            <p
              class="chakra-text css-63qbpm"
              data-testid="badge-text-testid"
            >
              Typescript
            </p>
          </div>
        </div>
        <div>
          <div
            style="transition: opacity 400ms, transform 400ms; transform: translateY(20px); opacity: 0;"
          >
            <p
              class="chakra-text css-kmn13v"
              data-testid="badge-text-testid"
            >
              CI/CD
            </p>
          </div>
        </div>
        <div>
          <div
            style="transition: opacity 400ms, transform 400ms; transform: translateY(20px); opacity: 0;"
          >
            <p
              class="chakra-text css-63qbpm"
              data-testid="badge-text-testid"
            >
              GraphQL
            </p>
          </div>
        </div>
        <div>
          <div
            style="transition: opacity 400ms, transform 400ms; transform: translateY(20px); opacity: 0;"
          >
            <p
              class="chakra-text css-kmn13v"
              data-testid="badge-text-testid"
            >
              Redux
            </p>
          </div>
        </div>
        <div>
          <div
            style="transition: opacity 400ms, transform 400ms; transform: translateY(20px); opacity: 0;"
          >
            <p
              class="chakra-text css-63qbpm"
              data-testid="badge-text-testid"
            >
              Jest
            </p>
          </div>
        </div>
        <div>
          <div
            style="transition: opacity 400ms, transform 400ms; transform: translateY(20px); opacity: 0;"
          >
            <p
              class="chakra-text css-kmn13v"
              data-testid="badge-text-testid"
            >
              HTML
            </p>
          </div>
        </div>
        <div>
          <div
            style="transition: opacity 400ms, transform 400ms; transform: translateY(20px); opacity: 0;"
          >
            <p
              class="chakra-text css-63qbpm"
              data-testid="badge-text-testid"
            >
              CSS
            </p>
          </div>
        </div>
        <div>
          <div
            style="transition: opacity 400ms, transform 400ms; transform: translateY(20px); opacity: 0;"
          >
            <p
              class="chakra-text css-kmn13v"
              data-testid="badge-text-testid"
            >
              JavaScript
            </p>
          </div>
        </div>
        <div>
          <div
            style="transition: opacity 400ms, transform 400ms; transform: translateY(20px); opacity: 0;"
          >
            <p
              class="chakra-text css-63qbpm"
              data-testid="badge-text-testid"
            >
              Android
            </p>
          </div>
        </div>
        <div>
          <div
            style="transition: opacity 400ms, transform 400ms; transform: translateY(20px); opacity: 0;"
          >
            <p
              class="chakra-text css-kmn13v"
              data-testid="badge-text-testid"
            >
              IOS
            </p>
          </div>
        </div>
        <div>
          <div
            style="transition: opacity 400ms, transform 400ms; transform: translateY(20px); opacity: 0;"
          >
            <p
              class="chakra-text css-63qbpm"
              data-testid="badge-text-testid"
            >
              Play Store
            </p>
          </div>
        </div>
        <div>
          <div
            style="transition: opacity 400ms, transform 400ms; transform: translateY(20px); opacity: 0;"
          >
            <p
              class="chakra-text css-kmn13v"
              data-testid="badge-text-testid"
            >
              Apple Store
            </p>
          </div>
        </div>
        <div>
          <div
            style="transition: opacity 400ms, transform 400ms; transform: translateY(20px); opacity: 0;"
          >
            <p
              class="chakra-text css-63qbpm"
              data-testid="badge-text-testid"
            >
              Maps API
            </p>
          </div>
        </div>
        <div>
          <div
            style="transition: opacity 400ms, transform 400ms; transform: translateY(20px); opacity: 0;"
          >
            <p
              class="chakra-text css-kmn13v"
              data-testid="badge-text-testid"
            >
              API REST
            </p>
          </div>
        </div>
        <div>
          <div
            style="transition: opacity 400ms, transform 400ms; transform: translateY(20px); opacity: 0;"
          >
            <p
              class="chakra-text css-63qbpm"
              data-testid="badge-text-testid"
            >
              NodeJS
            </p>
          </div>
        </div>
        <div>
          <div
            style="transition: opacity 400ms, transform 400ms; transform: translateY(20px); opacity: 0;"
          >
            <p
              class="chakra-text css-kmn13v"
              data-testid="badge-text-testid"
            >
              Styled Components
            </p>
          </div>
        </div>
        <div>
          <div
            style="transition: opacity 400ms, transform 400ms; transform: translateY(20px); opacity: 0;"
          >
            <p
              class="chakra-text css-63qbpm"
              data-testid="badge-text-testid"
            >
              Material UI
            </p>
          </div>
        </div>
        <div>
          <div
            style="transition: opacity 400ms, transform 400ms; transform: translateY(20px); opacity: 0;"
          >
            <p
              class="chakra-text css-kmn13v"
              data-testid="badge-text-testid"
            >
              Chakra UI
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
